app/src/org/commcare/dalvik/activities/EntitySelectActivity.java

```diff
@@ -572,12 +572,10 @@ protected void onActivityResult(int requestCode, int resultCode, Intent intent)
                 Callout callout = shortSelect.getCallout();
                 for (String key : callout.getResponses()) {
                     result = intent.getExtras().getString(key);
-                    if (result != null) {
-                        session.setDatum(key,result);
-                        if(!resultSet) {
-                            resultSet = true;
-                            this.searchbox.setText(result);
-                        }
+                    if (result != null && !resultSet) {
+                        resultSet = true;

```

Comment:
only because I noticed that the previous datum code was in here (when it shouldn't have been)